codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/codegen/StructureGenerator.java

```diff
@@ -289,18 +290,23 @@ private void renderErrorStructure() {
 
             // Write out a struct to hold the error data.
             writer.openBlock("type $L struct {", "}", structureSymbol.getName(), () -> {
-                // The message is the only part of the standard APIError interface that isn't known ahead of time.
-                // Message is a pointer mostly for the sake of consistency.
                 writer.write("$LBaseException", context.settings().getService().getName());
-                writer.write("Message *string").write("");
-                writer.write("ErrorCodeOverride *string").write("");
-
+                Set<String> memberNameSet = new HashSet<>();
                 for (MemberShape member : shape.getAllMembers().values()) {
                     String memberName = symbolProvider.toMemberName(member);
-                    // error messages are represented under Message for consistency
-                    if (!ERROR_MEMBER_NAMES.contains(memberName)) {
-                        writer.write("$L $P", memberName, symbolProvider.toSymbol(member));
-                    }
+                    memberNameSet.add(memberName);
+                    writer.write("$L $P", memberName, symbolProvider.toSymbol(member));
+                }
+                
+                // The message is the only part of the standard APIError interface that isn't known ahead of time.
+                // Message is a pointer mostly for the sake of consistency.
+
+                // If Message and ErrorCodeOverride is not defined in model.
+                if (!memberNameSet.contains("Message")) {
+                    writer.write("Message *string").write("");
+                }
+                if (!memberNameSet.contains("ErrorCodeOverride")) {
+                    writer.write("ErrorCodeOverride *string").write("");
```

Comment:
Why has this changed? This might break aws-models.

Reply:
This will break models with @required in message of error structure.

Reply:
We discussed, and Java's behavior would throw NPE but this case should never happen because aws-sdk will set the message to some value. Asked @rishav-karanjit to add a TODO.